### DISCUSSION TITLE
Is it possible to unmap "Shift+Space" or map it to something else?

### DISCUSSION THREAD

**Discussion by rempas** [kovidgoyal/kitty#7284] [Q&A] [answered]
Hello and thank you for your work and support!  I tried to learn and use kitty, and I have a very important problem:

I am using two custom keyboard layouts, [rempas](https://opengist.thomice.li/rempas/37f02c6807bd4e3e96863d17b5dc9fa9) and [rempas_el](https://opengist.thomice.li/rempas/2f14086f4b884a7088d1064f1bacb071) and in both of them, I have set "Shift+Space" to produce a backspace character. Kitty uses this to create a new prompt line (I don't know how to call it) and I tried map that to `\b` in the config file, but it didn't work.

Is there any way to change that behavior?

**Comment by kovidgoyal** [comment]
kitty does not create prompt lines. That would be the shell running
inside kitty. And you can unmap any keys you like in kitty, see the docs
for how to do that.

**Reply by rempas** [reply]
Hello and thank you for the quick reply! When I say, "prompt lines", I mean that it leaves a newline and adds the prompt again, the same way if you just pressed `Return (Enter)` and didn't type anything else (and the same way a process ends, and a new line is added, and a new prompt is created).

The thing is, this keybind is in the Waland/X11 level so shouldn't kitty as a program see a "backspace" when pressing "Shift+Return" unless, kitty somehow reads the physical input that you give it for these keys? And pretty much any other terminal emulator doesn't have any problems with that keybind, so I think kitty does something different.

**Reply by kovidgoyal** [reply, marked as answer]
kitty uses libxkbcommon to real all key input. If you want to remap keys you need to do it using something that works with libxkbcommon, whichis the standard way to process keys on linux. Run kitty with --debug-input and you will see what events it receives.

**Reply by rempas** [reply]
Thanks for the information! I run it the way you told me and it seems to be generating a "backspace" indeed but for some reason, it doesn't work as in every other program. In any case, I see that you cannot provide any more info so I don't want to bother you further. Thank you for your time and I wish you to have a lovely day!
